packages/kbn-ai-playground/components/summarization_panel/open_ai_summarization_model.tsx

```diff
@@ -0,0 +1,118 @@
+
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiIcon,
+  EuiSuperSelect,
+  EuiSuperSelectOption,
+  EuiToolTip
+} from "@elastic/eui";
+
+import { i18n } from "@kbn/i18n";
+
+import { OpenAIIcon } from "./open_ai_icon";
+
+const renderSelectOptions = (label: string) => (
+  <EuiFlexGroup alignItems="center">
+    <EuiFlexItem grow={false}>
+      <EuiIcon type={OpenAIIcon} />
+    </EuiFlexItem>
+    <EuiFlexItem>
+      {label}
+    </EuiFlexItem>
+  </EuiFlexGroup>
+);
+
+
+enum SummarizationModelName {
+  gpt3_5 = 'gpt-3.5-turbo',
+  gpt3_5_turbo_1106 = 'gpt-3.5-turbo-1106',
+  gpt3_5_turbo_16k = 'gpt-3.5-turbo-16k',
+  gpt3_5_turbo_16k_0613 = 'gpt-3.5-turbo-16k-0613',
+  gpt3_5_turbo = 'gpt-3.5-turbo-instruct',
+}
+
+const Summarization_Model: EuiSuperSelectOption<string>[] = [
+  {
+    value: SummarizationModelName.gpt3_5,
+    inputDisplay: renderSelectOptions(SummarizationModelName.gpt3_5),
+  },
+  {
+    value: SummarizationModelName.gpt3_5_turbo_1106,
+    inputDisplay: renderSelectOptions(SummarizationModelName.gpt3_5_turbo_1106),
+  },
+  {
+
+    value: SummarizationModelName.gpt3_5_turbo_16k,
+    inputDisplay: renderSelectOptions(SummarizationModelName.gpt3_5_turbo_16k),
+  },
+  {
+    value: SummarizationModelName.gpt3_5_turbo_16k_0613,
+    inputDisplay: renderSelectOptions(SummarizationModelName.gpt3_5_turbo_16k_0613),
+  },
+  {
+    value: SummarizationModelName.gpt3_5_turbo,
+    inputDisplay: renderSelectOptions(SummarizationModelName.gpt3_5_turbo),
+  },
+];
+
+interface OpenAISummarizationModelProps {
+  openAIFlyOutOpen: () => void;
+  model: string;
+  onSelect: (key: string) => void;
+}
+
+export const OpenAISummarizationModel: React.FC<OpenAISummarizationModelProps> = ({ model, onSelect, openAIFlyOutOpen }) => {
+
+  const [selectedModel, setSelectedModel] = useState(model ?? SummarizationModelName.gpt3_5_turbo_1106);
+
+  const onChange = (value: string) => {
+    setSelectedModel(value);
+    onSelect(value);
+  };
+
+  return (
+    <EuiFormRow
+      label={
+        <EuiToolTip
+          content={i18n.translate('aiPlayground.sidebar.summarizationModel.help', {
+            defaultMessage:
+              'The large language model used to summarize your documents.',
+          })}
+        >
+          <>
+            <span>
+              {i18n.translate('aiPlayground.sidebar.summarizationModel.label', {
+                defaultMessage: 'Summarization Model',
+              })}
+            </span>
+            <EuiIcon type="questionInCircle" color="subdued" />
+          </>
+        </EuiToolTip>
+      }
+      labelAppend={
+        <EuiButtonEmpty flush="both" size="xs" onClick={() => openAIFlyOutOpen()}>
+          {i18n.translate('aiPlayground.sidebar.summarizationModel.editLabel', {
+            defaultMessage: 'Edit OpenAI API key',
+          })}
+        </EuiButtonEmpty>
+      }
+    >
+      <EuiSuperSelect
+        options={Summarization_Model}
```

Comment:
Before I make the changes, I want to clarify if this is something is encourage in Elastic? 
The reason is that, under the hood, it is essentially the same thing. By creating a string array and render each element in the `options` or I can utilize `typescript` to create the array and plug in within the component.

Please let me know if you still feels strongly against it, I will push the changes or will change it in my next PR.

Reply:
It mostly to keep thing simple and avoid copypasting things and make it more complex and specific to our needs.
Anyway this solution is working so it up to you to apply or not this changes :)
Imagine we gonna add few more model, it will be just a new elements of array. If this elements will have another icon we don't need to create another render for it, we just will put icon as a new param and so on.

Reply:
got it